docs/add-merge-automation-to-your-repository.md

```diff
@@ -0,0 +1,38 @@
+# Adding merge automation to your repository
+
+In order to run tests and enable automated merging of pull requests for your repository on KubeVirt [prow] you need to have the following components configured:
+- [tide] - responsible for maintaining the merge pool
+- [lgtm] - sets `lgtm` label
+- approve - sets label `approved`
+- [trigger] - runs test jobs for a PR
+- [branch-protection] tide takes from it required status for the [presubmit] jobs configured for the repository
+
+In a nutshell: if tide is enabled then for any created PR that has `lgtm` and `approved` labels and all presubmit checks that are required are successful, tide will merge it. 
+
+Example configuration PR to enable tide for KubeVirt [prow]: https://github.com/kubevirt/project-infra/pull/653/
+
+Example workflow:
+- tide is configured for PRs on a repository to require `lgtm` and `approved` labels and missing `needs-rebase` label
+- Alice creates a pull request, which triggers presubmit jobs to run for the PR
+- some presubmit jobs have `optional: false` set, which makes their status required to be successful
+- All jobs succeed and add a success status to the commit object in GitHub
+- Note: tide does not consider this PR to be in the merge pool as it misses the `lgtm` and `approved` labels
+- Alice asks Bob and Charles for a review on the pull request
+- Bob (a member) reviews and adds a `/lgtm` comment
+- After that the default branch gets updated by another merge, where merging Alice's PR would cause a conflict, which adds a `needs-rebase` label
+- Charles (a maintainer) reviews and adds an `/approve` comment
+- Note: the `approved` label added to the PR enables automatic retesting on the PR
```

Comment:
Even if it has `needs-rebase`? Why do we retest on approve anyway?

Reply:
I think it refers to our `/retest` bot, it only sets the retest comment on approved PRs.

Reply:
TBH that's a very good point, maybe we should add that `needs-rebase` label at the periodic job definition: https://github.com/kubevirt/project-infra/blob/97ce8b6cc7bf8c66c58e02f47c1ce31e580c8181/github/ci/prow-deploy/files/jobs/kubevirt/project-infra/project-infra-periodics.yaml#L17

WDYT?

Reply:
Sounds very good :+1:

Reply:
Oh, I just saw it's already there: https://github.com/kubevirt/project-infra/blob/97ce8b6cc7bf8c66c58e02f47c1ce31e580c8181/github/ci/prow-deploy/files/jobs/kubevirt/project-infra/project-infra-periodics.yaml#L26

Reply:
So jed, you are right, as long as the need-rebase is present, no retesting by the bot will occur